### ISSUE TITLE
How to execute simple SQL

### ISSUE THREAD

**Issue by wangduanqing5945** [greenrobot/greenDAO#409]
"select t.name,t.age from user t where t.age > 10"、
"update user set age=20 where name='Jackson'"
in greenDao like this simple SQL，how to do with effectively!  I don't want to update JavaObject all property。Or, I want to select two or three property, not all of this!  
Who can help me?

**Comment by greenrobot-team** [comment]
There is only a very minor performance impact if you read all properties (`dao.load()` and `dao.queryBuilder()`).

So just query for the `user` objects you want to update, change them, then save them back to the database using `dao.updateInTx(users)`.

The purpose of greenDAO is that you can simply use Java objects instead of having to use SQL like you suggested.
-ut

**Comment by PeeJWeeJ** [comment]
Isn't this implicitly supported by already relying on SQLiteDatabase? If you really want to do direct SQL, just use the SQLiteDatabase you use to create your DaoMaster

**Comment by greenrobot-team** [comment]
@PeeJWeeJ Yes, you can. But @wangduanqing5945 asked how to do this with greenDAO, so... :)

**Comment by wangduanqing5945** [comment]
@PeeJWeeJ  more than one hundred properties. I think there is not minor performance.  but ,thank you for reply

**Comment by PeeJWeeJ** [comment]
@wangduanqing5945 What are you doing that requires specific optimization? You can build pretty complex queries with QueryBuilder and (I believe) that is simply translated into the SQL call and then executed

**Comment by wangduanqing5945** [comment]
@PeeJWeeJ  Yes，I finish the optimization. I packeted complex update/simple select/count method. Thanks！But,I find out another issue in SqlUtils. appendColumnsEqValue method，append with "," not "and", this is a bug?   if not, the sql will be "update user set name='aa' where id=?,name=?,sex=?"

**Comment by wangduanqing5945** [comment]
@PeeJWeeJ  here
![image](https://cloud.githubusercontent.com/assets/5267198/17925916/75fb96e8-6a21-11e6-9e7c-bbe4921dad44.png)

**Comment by greenrobot-team** [comment]
@wangduanqing5945 Alright, yes, this is likely incorrect. However, this has no consequences for greendao as only a single primary key on an entity is supported.

You should not use `SqlUtils` in your code. It is for internal use by greendao only.
-ut
